### ISSUE TITLE
`WITH ... DELETE`, which has `[]byte` arguments errors with `Invalid UTF-8`.

### ISSUE THREAD

**Issue by egonelbre** [cockroachdb/cockroach#63818]
**Describe the problem**

Executing the following query results in `error in argument for $1: invalid UTF-8 sequence (SQLSTATE 22021)`:

```
_, err = conn.Exec(ctx, `
	WITH y AS (
		DELETE FROM alpha
		WHERE key = $1::BYTEA AND sid = $2::BYTEA
		RETURNING 1
	)
	DELETE FROM beta
	WHERE sid = $2::BYTEA
`, []byte{0xFF}, []byte{0xFF})
```

The query works fine on postgresql. The problem happens with both `github.com/lib/pq` and `github.com/jackc/pgx/v4` drivers.

**To Reproduce**

1. Setup a new node with `cockroach start-single-node`.
2. Execute the following code:

``` go
package main

import (
	"context"
	"database/sql"
	"flag"

	"github.com/jackc/pgx/v4"
	_ "github.com/lib/pq"
)

func main() {
	connstr := flag.String("db", "postgresql://root@localhost:26257?sslmode=disable", "database connection")
	pq := flag.Bool("pq", false, "use lib/pq driver")
	flag.Parse()

	ctx := context.Background()
	var mustExec func(ctx context.Context, query string, args ...interface{})

	if !*pq {
		conn, err := pgx.Connect(ctx, *connstr)
		check(err)
		defer conn.Close(ctx)

		mustExec = func(ctx context.Context, query string, args ...interface{}) {
			_, err := conn.Exec(ctx, query, args...)
			if err != nil {
				panic(err)
			}
		}
	} else {
		db, err := sql.Open("postgres", *connstr)
		check(err)
		defer db.Close()

		mustExec = func(ctx context.Context, query string, args ...interface{}) {
			_, err := db.ExecContext(ctx, query, args...)
			if err != nil {
				panic(err)
			}
		}
	}

	mustExec(ctx, `
		DROP TABLE IF EXISTS alpha;
		DROP TABLE IF EXISTS beta;
		CREATE TABLE alpha (
			key  BYTEA NOT NULL,
			sid  BYTEA NOT NULL,
			PRIMARY KEY (key)
		);
		CREATE TABLE beta (
			sid  BYTEA NOT NULL,
			PRIMARY KEY (sid)
		)
	`)

	mustExec(ctx, `
		WITH y AS (
			DELETE FROM alpha
			WHERE key = $1::BYTEA AND sid = $2::BYTEA
			RETURNING 1
		)
		DELETE FROM beta
		WHERE sid = $2::BYTEA
	`, []byte{0xFF}, []byte{0xFF})
}

func check(err error) {
	if err != nil {
		panic(err)
	}
}
```

**Environment:**
 - CockroachDB version: build: CCL v20.2.7 @ 2021/03/29 17:56:44 (go1.13.14)
 - Server OS: macOS Big Sur
 - Client app: `github.com/lib/pq` and `github.com/jackc/pgx/v4`

**Comment by rafiss** [comment]
Thanks for this report @egonelbre ! We'll take a closer look soon

**Comment by rafiss** [comment]
Actually, just took a look now :)

This is fixed in CockroachDB 21.1, which will be released soon. You can try out a beta version of it in the meantime: https://www.cockroachlabs.com/docs/releases/#testing-releases

The fix was https://github.com/cockroachdb/cockroach/pull/60949/ 

@egonelbre I'll close this, but let me know if you have any questions!
